model/distro/distro.go

```diff
@@ -876,3 +877,49 @@ func AllDistros(ctx context.Context) ([]Distro, error) {
 func AllDistroIDs(ctx context.Context) ([]Distro, error) {
 	return Find(ctx, bson.M{}, options.Find().SetSort(bson.M{IdKey: 1}).SetProjection(bson.M{IdKey: 1}))
 }
+
+type DistroSettingsSection string
+
+const (
+	DistroSettingsGeneral  DistroSettingsSection = "GENERAL"
+	DistroSettingsProvider DistroSettingsSection = "PROVIDER"
+	DistroSettingsTask     DistroSettingsSection = "TASK"
+	DistroSettingsHost     DistroSettingsSection = "HOST"
+	DistroSettingsProject  DistroSettingsSection = "PROJECT"
+)
+
+// UpdateDistroSection saves the changes for a given section and logs the events with the associated user.
+func UpdateDistroSection(ctx context.Context, originalDistro *Distro, changes *Distro, section DistroSettingsSection, userID string) (*Distro, error) {
```

Comment:
It might be tricky to combine them at this point since only one section (the general section) is complete and there may be some unexpected curve balls down the road -- we could definitely revisit it though!